### PR TITLE
feat(route): Add GitHub Notification

### LINKED ISSUE #12042: 请求添加gh的notification的rss
### 类型

编程

### 网站地址

https://github.com/notifications

### 网站描述

看到有很多可以自建的“gh通知转rss”服务（比如 [skynewz/github-notifications-rss Tags | Docker Hub](https://hub.docker.com/r/skynewz/github-notifications-rss/tags)）但是懒得自建，如果能集成到rsshub就太好了。

### 需要生成什么内容？

每条通知的内容

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by zhzy0077**
## 该 PR 相关 Issue / Involved issue

Close #12042

## 路由地址示例 / Example for the proposed route(s)
```routes
/github/notifications
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [X] 新的路由 New Route
  - [X] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [X] 文档说明 Documentation
  - [X] 中文文档 CN
  - [X] 英文文档 EN
- [ ] 全文获取 fulltext
  - [ ] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [ ] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

## 说明 / Note
The output is looking like:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<rss  xmlns:atom="http://www.w3.org/2005/Atom" version="2.0">
    <channel>
        <title>
            <![CDATA[Github Notifications]]>
        </title>
        <link>https://github.com/notifications</link>
        <atom:link href="http://localhost:1200/github/notifications" rel="self" type="application/rss+xml" />
        <description>
            <![CDATA[Github Notifications - Made with love by RSSHub(https://github.com/DIYgod/RSSHub)]]>
        </description>
        <generator>RSSHub</generator>
        <webMaster>i@diygod.me (DIYgod)</webMaster>
        <language>zh-cn</language>
        <lastBuildDate>Wed, 29 Mar 2023 11:09:08 GMT</lastBuildDate>
        <ttl>5</ttl>
        <item>
            <title>
                <![CDATA[New version: clouDr-f2e.rubick version 2.1.6]]>
            </title>
            <description>
                <![CDATA[New version: clouDr-f2e.rubick version 2.1.6]]>
            </description>
            <pubDate>Wed, 29 Mar 2023 11:06:19 GMT</pubDate>
            <guid isPermaLink="false">6000769352</guid>
            <link>https://api.github.com/repos/microsoft/winget-pkgs/pulls/100983</link>
        </item>
    </channel>
</rss>
```

### CODE REVIEW

File: lib/v2/github/notifications.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const got = require('@/utils/got');
+const { parseDate } = require('@/utils/parse-date');
+
+const apiUrl = 'https://api.github.com';
+const config = require('@/config').value;
+
+module.exports = async (ctx) => {
+    const headers = {
```
Comment (TonyRL) [inline lib/v2/github/notifications.js:8]:
```suggestion
    if (!config.github || !config.github.access_token) {
        throw Error('GitHub trending RSS is disabled due to the lack of <a href="https://docs.rsshub.app/install/#pei-zhi-bu-fen-rss-mo-kuai-pei-zhi">relevant config</a>');
    }
    const headers = {
```

File: lib/v2/github/router.js [hunk 2]
```diff
@@ -14,4 +14,5 @@ module.exports = function (router) {
     router.get('/trending/:since/:language/:spoken_language?', require('./trending'));
     router.get('/user/followers/:user', require('./follower'));
     router.get('/wiki/:user/:repo/:page?', require('./wiki'));
+    router.get('/notifications', require('./notifications'));
```
Comment (TonyRL) [inline lib/v2/github/router.js:17]: Please insert new entries in alphabetic order.

**Review by TonyRL — COMMENTED** [review]
Missing `maintainer.js`

**Comment by zhzy0077** [comment]
> Missing `maintainer.js`

Sorry I was not aware of that file. Updated accordingly

**Pull request merged**